langchain/vectorstores/opensearch_vector_search.py

```diff
@@ -0,0 +1,366 @@
+                "text": text,
+                "metadata": metadata,
+                "_id": _id,
+            }
+            ids.append(_id)
+            requests.append(request)
+        bulk(self.client, requests)
+        if refresh_index:
+            self.client.indices.refresh(index=self.index_name)
+        return ids
+
+    def similarity_search(
+        self, query: str, k: int = 4, **kwargs: Any
+    ) -> List[Document]:
+        """Return docs most similar to query.
+
+        By default supports Approximate Search.
+        Also supports Script Scoring and Painless Scripting.
+
+        Args:
+            query: Text to look up documents similar to.
+            k: Number of Documents to return. Defaults to 4.
+
+        Returns:
+            List of Documents most similar to the query.
+
+        Optional Args for Approximate Search:
+            search_type: "approximate_search"; default: "approximate_search"
+            size: number of results the query actually returns; default: 10000
+
+        Optional Args for Script Scoring Search:
+            search_type: "script_scoring"; default: "approximate_search"
+
+            space_type: "l2", "l1", "linf", "cosinesimil", "innerproduct",
+            "hammingbit"; default: "l2"
+
+            pre_filter: script_score query to pre-filter documents before identifying
+            nearest neighbors; default: {"match_all": {}}
+
+        Optional Args for Painless Scripting Search:
+            search_type: "painless_scripting"; default: "approximate_search"
+            space_type: "l2Squared", "l1Norm", "cosineSimilarity"; default: "l2Squared"
+
+            pre_filter: script_score query to pre-filter documents before identifying
+            nearest neighbors; default: {"match_all": {}}
+        """
+        embedding = self.embedding_function.embed_query(query)
+        search_type = _get_kwargs_value(kwargs, "search_type", "approximate_search")
+        if search_type == "approximate_search":
+            size = _get_kwargs_value(kwargs, "size", 10000)
+            search_query = _default_approximate_search_query(embedding, size, k)
+        elif search_type == SCRIPT_SCORING_SEARCH:
+            space_type = _get_kwargs_value(kwargs, "space_type", "l2")
+            pre_filter = _get_kwargs_value(kwargs, "pre_filter", MATCH_ALL_QUERY)
+            search_query = _default_script_query(embedding, space_type, pre_filter)
+        elif search_type == PAINLESS_SCRIPTING_SEARCH:
+            space_type = _get_kwargs_value(kwargs, "space_type", "l2Squared")
+            pre_filter = _get_kwargs_value(kwargs, "pre_filter", MATCH_ALL_QUERY)
+            search_query = _default_painless_scripting_query(
+                embedding, space_type, pre_filter
+            )
+        else:
+            raise ValueError("Invalid `search_type` provided as an argument")
+
+        response = self.client.search(index=self.index_name, body=search_query)
```

Comment:
Should we exclude returning the vector in the search?

Reply:
That's the same logic used in all vector stores and is used by end users. So, I followed the same.